tooling/nargo/Cargo.toml

```diff
@@ -21,20 +21,26 @@ noirc_errors.workspace = true
 noirc_frontend.workspace = true
 noirc_printable_type.workspace = true
 iter-extended.workspace = true
+rayon.workspace = true
 thiserror.workspace = true
 tracing.workspace = true
-rayon.workspace = true
-jsonrpc.workspace = true
-rand.workspace = true
-serde.workspace = true
 walkdir = "2.5.0"
 
+# Some dependencies are optional so we can compile to Wasm.
+serde = { workspace = true, optional = true }
+jsonrpsee = { workspace = true, optional = true }
+tokio = { workspace = true, optional = true }
+rand = { workspace = true, optional = true }
+
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-noir_fuzzer.workspace = true
-proptest.workspace = true
+noir_fuzzer = { workspace = true }
+proptest = { workspace = true }
 
 [dev-dependencies]
-jsonrpc-http-server = "18.0"
-jsonrpc-core-client = "18.0"
-jsonrpc-derive = "18.0"
-jsonrpc-core = "18.0"
+jsonrpsee = { workspace = true, features = ["server"] }
+
+[features]
+default = ["rpc"]
```

Comment:
Can we make this a non-default flag and have `nargo_cli` turn it on? This should allow us to have `compiler/wasm` use the workspace dependency no? (maybe you tried this already.

Reply:
I haven’t tried that, no. I just wanted to keep it backwards compatible for any potential user who already have this as a dependency, and have the override for noir_wasm, hidden in our build. We can try, I don’t have a strong opinion either way.

Reply:
Yes, I can confirm that it would work that way. The following need to add the `rpc` feature:
* `/tooling/acvm_cli/Cargo.toml`
* `/tooling/debugger/Cargo.toml`
* `/tooling/profiler/Cargo.toml`
Not sure why `nargo_cli` didn't complain 🤔 

Up to you, like I say I didn't want to disrupt anyone who is used to having access to the `DefaultForeignCallExecutor`, which is not available without `rpc`.

Reply:
Ah so the workspace unification is still in play: as soon as one crate required the `rpc` feature it became the default, which is why `nargo_cli` didn't need to specify it. It's weird that the Wasm crate even compiled 😵 

I was trying to figure out a way where we can keep the `DefaultForeignCallExecutor` but change its API based on whether the RPC config is accepted or not. For example the `acvm-cli` and `profiler` crates don't pass any URL, and it's not really enabled in the `debugger` crate either.

Reply:
Changed it so `rpc` is an opt-in feature.